hpx/traits/serialization_access_data.hpp

```diff
@@ -96,19 +96,20 @@ namespace hpx { namespace traits
             void* dest = &cont[current];
             switch (count)
             {
+            case 16:
+                std::memcpy(dest, address, 16);
+                break;
+
             case 8:
-                *static_cast<std::uint64_t*>(dest) =
-                    *static_cast<std::uint64_t const*>(address);
+                std::memcpy(dest, address, 8);
                 break;
 
             case 4:
-                *static_cast<std::uint32_t*>(dest) =
-                    *static_cast<std::uint32_t const*>(address);
+                std::memcpy(dest, address, 4);
                 break;
 
             case 2:
-                *static_cast<std::uint16_t*>(dest) =
-                    *static_cast<std::uint16_t const*>(address);
+                std::memcpy(dest, address, 2);
```

Comment:
Anyway, ```std::memcpy``` is performed in ```default:```. So, as I think, there is no need to be handled separately.

Reply:
The point of repeating the `memcpy()` invocations with constant byte sizes is that compilers usually optimize those, whereas the generic `memcpy(d, s, N)` can't be optimized as aggressively.

Reply:
@hkaiser Oh, makes sense! I got it 👍 You're right.

Reply:
Does the benefit of compiler optimizations exceed the overhead of switch-case statement?

Reply:
The measurements show that serialization is up to 10% faster with this change.